ApiReview/API.Polly.Core/NoDocs/Polly/ResilienceProperties.cs

```diff
@@ -0,0 +1,17 @@
+// Assembly 'Polly.Core'
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Polly;
+
+public sealed class ResilienceProperties : IDictionary<string, object?>, ICollection<KeyValuePair<string, object?>>, IEnumerable<KeyValuePair<string, object?>>, IEnumerable
+{
+    object? IDictionary<string, object>.this[string key] { get; set; }
+    public bool TryGetValue<TValue>(ResiliencePropertyKey<TValue> key, [MaybeNullWhen(false)] out TValue value);
+    public TValue GetValue<TValue>(ResiliencePropertyKey<TValue> key, TValue defaultValue);
+    public void Set<TValue>(ResiliencePropertyKey<TValue> key, TValue value);
```

Comment:
One common use-case of V7 is attaching primitive types (see #890 as an example). While in V8 they could use something like `properties.Set<int>(3)` or `properties.Set<MyIntWrapper>()` it could lead to ambiguity or more more complicated usage.